ginivision/src/main/java/net/gini/android/vision/review/multipage/thumbnails/ThumbnailsAdapter.java

```diff
@@ -317,14 +346,35 @@ public void setItemTouchHelper(final ItemTouchHelper itemTouchHelper) {
         mItemTouchHelper = itemTouchHelper;
     }
 
+    public void setUploadState(final UploadState uploadState,
+            @NonNull final ImageDocument document) {
+        for (int i = 0; i < mMultiPageDocument.getDocuments().size(); i++) {
+            final ImageDocument imageDocument = mMultiPageDocument.getDocuments().get(i);
+            if (imageDocument.equals(document)) {
+                final Thumbnail thumbnail = mThumbnails.get(i);
+                thumbnail.uploadState = uploadState;
+                notifyItemChanged(i);
+                break;
+            }
+        }
+    }
+
     enum ViewType {
         THUMBNAIL,
         PLUS_BUTTON
     }
 
+    public enum UploadState {
+        NOT_STARTED,
```

Comment:
Does it make sense to have this state? I mean, the initial state should always be `IN_PROGRESS`

Reply:
From the view's point of view it makes sense for me. The default state for it is NOT_STARTED to not make assumptions and it shows `IN_PROGRESS` only when the upload is really in progress.